backend/src/target.ts

```diff
@@ -44,53 +42,55 @@ class Target {
 
       if (isExtrinsicFailed) {
         this.logger.error("Extrinsic failed");
+      } else {
+        this.logger.debug(
+          `Transaction included: ${polkadotAppsUrl}${status.asInBlock}`
+        );
       }
-
-      this.logger.debug(
-        `Transaction included: ${polkadotAppsUrl}${status.asInBlock}`
-      );
+      subscription.unsubscribe();
```

Comment:
I am not sure we need to explicitly call `unsubscribe` here, because we're using `takeWhile(({ status }) => !status.isInBlock, true)`, which should unsubscribe once tx status is in block

Reply:
https://medium.com/javascript-everyday/no-need-to-unsubscribe-rxjs-operators-will-help-you-out-f8b8ce7bf26a

Reply:
Without this it leaks memory like crazy, with this memory leak stopped, I'm running relayer for over an hour with 500 blocks/s speed and nothing leaks.

It is indeed necessary to always unsubscribe and even documentation contains that in examples.

Reply:
But you call `.subscribe()` after `takeWhile()`, so that subscription isn't handled.

Reply:
`.subscribe()` always comes last, after operators. if that is indeed the case maybe we should get rid of `takeWhile(({ status }) => !status.isInBlock, true)`

Reply:
Maybe, it is still confusing to me how it works. Try this change locally, you should see the same behavior.